### PR TITLE
feat: automatic generation of table of contents

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR
Screenshot 2021-12-21 at 12 53 45

Table of Content is generated automatically for each MDX file.

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -1,98 +1,5 @@
 # Migration
 
-## Table of contents
```
Comment (bgutsol) [inline MIGRATION.md:3]: If I'm not mistaken, this file can be viewed only within Github, there is no related page on the website. So I'd keep the TOC here.

File: packages/website/components/TableOfContent.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    marginBottom: tokens.spacingL,
+    padding: tokens.spacingL,
+    marginTop: `calc(-1 * ${tokens.spacingL})`,
+    '@media (min-width: 1400px)': {
+      width: '350px',
+      marginLeft: '50px',
+      marginRight: '-100px',
+      display: 'inline-block',
+      background: 'white',
+      float: 'right',
```
Comment (bgutsol) [inline packages/website/components/TableOfContent.tsx:20]: float ❤️
At first glance on the website, I couldn't get how you did that magic with layout. But then I found that lovely `float` 😄 

I guess, in near future, we gonna make this TOC sticky, so it stays with scroll, right?

**Review by bgutsol — APPROVED** [review]
👏

File: MIGRATION.md (hunk 1 above)
Comment (suevalov) [reply MIGRATION.md:3]: ah, good point

File: packages/website/components/TableOfContent.tsx (hunk 2 above)
Comment (suevalov) [reply packages/website/components/TableOfContent.tsx:20]: yeah, we can make it sticky later

Comment (suevalov) [reply packages/website/components/TableOfContent.tsx:20]: old school tricks 👴🏼 🪄

Comment (burakukula) [reply packages/website/components/TableOfContent.tsx:20]: WOW 😮

**Review by burakukula — APPROVED** [review]
👏

**Pull request merged**